shpc/main/registry/__init__.py

```diff
@@ -0,0 +1,209 @@
+__author__ = "Vanessa Sochat"
+__copyright__ = "Copyright 2021-2022, Vanessa Sochat"
+__license__ = "MPL 2.0"
+
+
+import os
+import shutil
+
+import jsonschema
+
+import shpc.main.schemas
+import shpc.utils
+from shpc.logger import logger
+from shpc.main.settings import SettingsBase
+
+from .filesystem import Filesystem, FilesystemResult
+from .remote import GitHub, GitLab
+
+
+def update_container_module(module, from_path, existing_path):
+    """
+    Update a container module, meaning copying over all files
+    """
+    if not os.path.exists(existing_path):
+        shpc.utils.mkdir_p(existing_path)
+    for filename in shpc.utils.recursive_find(from_path):
+        relative_path = filename.replace(from_path, "").strip("/")
+        to_path = os.path.join(existing_path, relative_path)
+        if os.path.exists(to_path):
+            shutil.rmtree(to_path)
+        shpc.utils.mkdir_p(os.path.dirname(to_path))
+        shutil.copyfile(filename, to_path)
```

Comment:
This is where the copy happens. According to the docs, this only copies the content of the file, not its mode or other stats. That's why it dropped the executable bits from `vanessa/salad/*_fork.sh`.
The simplest solution (tested) is to use the full-blown `copy`, which copies everything:
```suggestion
        shutil.copy(filename, to_path)
```